packages/core/src/extensions/install-extension/install-extension.injectable.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { fork } from "child_process";
+import AwaitLock from "await-lock";
+import pathToNpmCliInjectable from "../../common/app-paths/path-to-npm-cli.injectable";
+import extensionPackageRootDirectoryInjectable from "./extension-package-root-directory.injectable";
+import prefixedLoggerInjectable from "../../common/logger/prefixed-logger.injectable";
+import readJsonFileInjectable from "../../common/fs/read-json-file.injectable";
+import joinPathsInjectable from "../../common/path/join-paths.injectable";
+import type { PackageJson } from "../common-api";
+import writeJsonFileInjectable from "../../common/fs/write-json-file.injectable";
+import { once } from "lodash";
+
+const baseNpmInstallArgs = [
+  "install",
+  "--save-optional",
+  "--audit=false",
+  "--fund=false",
+  // NOTE: we do not omit the `optional` dependencies because that is how we specify the non-bundled extensions
+  "--omit=dev",
+  "--omit=peer",
+  "--prefer-offline",
+];
+
+export type InstallExtension = (name: string) => Promise<void>;
+
+const installExtensionInjectable = getInjectable({
+  id: "install-extension",
+  instantiate: (di): InstallExtension => {
+    const pathToNpmCli = di.inject(pathToNpmCliInjectable);
+    const extensionPackageRootDirectory = di.inject(extensionPackageRootDirectoryInjectable);
+    const readJsonFile = di.inject(readJsonFileInjectable);
+    const writeJsonFile = di.inject(writeJsonFileInjectable);
+    const joinPaths = di.inject(joinPathsInjectable);
+    const logger = di.inject(prefixedLoggerInjectable, "EXTENSION-INSTALLER");
+
+    const forkNpm = (...args: string[]) => new Promise<void>((resolve, reject) => {
+      const child = fork(pathToNpmCli, args, {
+        cwd: extensionPackageRootDirectory,
+        silent: true,
+        env: {},
+      });
+      let stderr = "";
+
+      child.stderr?.on("data", data => {
+        stderr += String(data);
+      });
+
+      child.on("close", (code) => {
+        if (code !== 0) {
+          reject(new Error(stderr));
+        } else {
+          resolve();
+        }
+      });
+
+      child.on("error", error => {
+        reject(error);
+      });
+    });
+
+    const packageJsonPath = joinPaths(extensionPackageRootDirectory, "package.json");
+
+    const fixupPackageJson = once(async () => {
+      const packageJson = await readJsonFile(packageJsonPath) as PackageJson;
+
+      delete packageJson.dependencies;
```

Comment:
This needs a comment why we're doing it for future reference

Reply:
It seems in current stable 3rd party extensions are `dependencies`, so this would delete these also. Will this cause problems?

```
➜ cat package.json
{
  "dependencies": {
    "@mirantis/lens-extension-cc": "file:../../../.k8slens/extensions/mirantis--lens-extension-cc",
```

Reply:
Maybe no, because the installed extension is still in `/Users/<username>/Library/Application Support/Lens/node_modules/` and `/Users/phorsmalahti/Library/Application Support/Lens/lens-extensions.json`. But doesn't sound good anyway.

Reply:
For consistency we could leave out `--save-optional` if we really don't need the exts in the package.json file.

Reply:
I'll add a comment. I was trying to emulate what was removed in #6715 since that is when this was all changed.

Reply:
Yes that is correct